embulk-core/src/main/java/org/embulk/exec/PreviewExecutor.java

```diff
@@ -174,10 +176,15 @@ public void add(Page page)
         @Override
         public void finish()
         {
+            if (res != null) {
+                // if PreviewResult object already exists, we should reuse it and throw PreviewedNoticeError.
```

Comment:
Added logging on 7ac9423d4d84a073ed2cbd6e795b2dd5707e3bc0. Plugin developers might be able to know the place where `PageOutput#finish` is called twice in the plugin and how to fix it.